docs/Gopkg.toml.md

```diff
@@ -0,0 +1,104 @@
+# Gopkg.toml Documenation
+
+## `[metadata]`
+metadata can exist at the root of the manifest as well as under `constraint` and `override`. 
```

Comment:
Also, we can move

>The metadata defined here will be ignored by dep.

lines below for two different cases (root and constraint & override), right after the first sentence and avoid repeating.

> `metadata` is always ignored by dep.